ql/src/java/org/apache/hadoop/hive/ql/optimizer/DynamicPartitionPruningOptimization.java

```diff
@@ -319,8 +319,9 @@ private boolean disableSemiJoinOptDueToExternalTable(HiveConf conf, TableScanOpe
           if (columnOrigin != null && columnOrigin.op instanceof TableScanOperator) {
             // Join key origin has been traced to a table column. Check if the table is external.
             TableScanOperator joinKeyTs = (TableScanOperator) columnOrigin.op;
-            if (MetaStoreUtils.isExternalTable(joinKeyTs.getConf().getTableMetadata().getTTable())) {
-              LOG.debug("Join key {} is from {} which is an external table. Disabling semijoin optimization.",
+            if (!StatsUtils.checkCanProvideStats(new Table(joinKeyTs.getConf().getTableMetadata().getTTable()))) {
+              LOG.debug("Join key {} is from {} which is an external table and also could not provide statistics. " +
```

Comment:
I know It was in the previous version as well, but inside the { } should not be a string for the table name?

Reply:
It is a string placeholder for the table name indeed.  Table name value is here line  `L325 `https://github.com/apache/hive/pull/4395/files#diff-1ade8830fed6939060483b84496a26eecab0e7ef64e19265a8e1102e1178fea2L325

Reply:
ACK In C# we have different style and it confused me :)